src/Services/AuthServices.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+
+namespace QuizApp\Services;
+
+use QuizApp\Entities\User;
+use QuizApp\Exceptions\UserNotFoundException;
+
+class AuthServices extends AbstractServices
+{
+    public function authenticate(string $email, string $password)
+    {
+        $filters = ['email' => $email, 'password' => md5($password)];
+        $user =  $this->repoManager->getRepository(User::class)->findOneBy($filters);
+        if (!$user){
+            $this->session->set('errorMessage', 'Email or Password incorrect !');
+            throw new UserNotFoundException();
+        }
+        $this->session->set('id', $user->getId());
+        $this->session->set('name', $user->getName());
+        $this->session->set('email', $user->getEmail());
+        $this->session->set('role', $user->getRole());
+
+        return $user;
+    }
+
```

Comment:
As discussed It should not be the responsibility of the auth service to retrieve the error message. 
You can work with the session directly both here and in the controller in the first iteration and afterwords add a wrapper in a further iteration smth like MessageManager or AppMessageService which handles the storage and retrieval of your error/success/warning messages.